x-pack/legacy/plugins/apm/public/components/shared/LocalFilters/Filter/index.tsx

```diff
@@ -0,0 +1,179 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React, { useState, useCallback, useMemo, useEffect } from 'react';
+import {
+  EuiTitle,
+  EuiButtonEmpty,
+  EuiPopover,
+  EuiSelectable,
+  EuiSpacer,
+  EuiHorizontalRule,
+  EuiText,
+  EuiButton,
+  EuiFlexItem,
+  EuiFlexGroup
+} from '@elastic/eui';
+import { i18n } from '@kbn/i18n';
+import theme from '@elastic/eui/dist/eui_theme_light.json';
+import styled from 'styled-components';
+import { FilterBadgeList } from './FilterBadgeList';
+
+const Popover = styled(EuiPopover).attrs({
+  anchorClassName: 'anchor'
+})`
+  .anchor {
+    display: block;
+  }
+`;
+
+const SelectContainer = styled.div`
+   {
+    width: 300px;
+  }
+`;
+
+const Counter = styled.div`
+  border-radius: ${theme.euiBorderRadius};
+  background: ${theme.euiColorLightShade};
+  padding: 0 ${theme.paddingSizes.xs};
+`;
+
+interface Props {
+  name: string;
+  title: string;
+  options: Array<{
+    name: string;
+    count: number;
+  }>;
+  onChange: (value: string[]) => void;
+  value: string[];
+}
+
+const Filter: React.FC<Props> = props => {
+  const { name, title, options, onChange, value } = props;
+
+  const [showPopover, setShowPopover] = useState(false);
+
+  const toggleShowPopover = useCallback(() => setShowPopover(!showPopover), [
```

Comment:
Is `useCallback` necessary here? Unless we run into perf issues we shouldn't add it. Same with `useMemo`.

I also saw it in https://github.com/elastic/kibana/blob/376240d81b4d1a0c968bb8382519e5125c81fd10/x-pack/legacy/plugins/apm/public/components/shared/ManagedTable/index.tsx where `useMemo` and `useCallback`  is now used several places.

Did we profile and found a big perf gain by doing so? And was there an issue to begin with?

The only place I've found it to noticable make a difference is for the charts since they are expensive to render. And this only becomes a problem when the user hovers over them quickly.

Reply:
Yeah I think I'm just defaulting to memoizations because the app just re-renders way too often. I don't think it's particularly important in this case. I'll remove it. (Another good point for bringing in the RUM agent would be to have some data points about how much time we spend re-rendering the application).